index.js

```diff
@@ -1,31 +1,29 @@
-const String = new Proxy({}, {
-  get: (_, name) => name
-})
+/**
+ * @template Value
+ * @param {(name: string) => Value} mapper
+ * @returns {{ [name: string]: Value }}
+ */
+const Values = (mapper) =>
+  new Proxy(
+    {},
+    {
+      /** @param {string} name */
+      get: (_, name) => mapper(name),
+    }
+  )
 
-const StringLower = new Proxy({}, {
-  get: (_, name) => name.toLowerCase()
-})
+const Strings = Values((name) => name)
 
-const Numeric = (() => {
-  const arr = []
-  
-  return new Proxy({}, {
-    get: (_, name) => {
-      arr.push(name)
-      return arr.indexOf(name)
-    }
-  })
-})()
+const LowerCased = Values((name) => name.toLowerCase())
 
-const NumericAt = (startIndex) => {
-  const arr = [...Array(startIndex).keys()]
-  
-  return new Proxy({}, {
-    get: (_, name) => {
-      arr.push(name)
-      return arr.indexOf(name)
-    }
+const Symbols = Values(Symbol)
+
+const Integers = (startIndex = 0) => {
+  const indexOfName = new Map()
+  return Values((name) => {
+    indexOfName.set(name, indexOfName.size + startIndex)
+    return indexOfName.get(name)
   })
```

Comment:
Golfing: `Map.set()` returns the `Map` object, so you could chain `.set()` and `.get()` and use implicit return. 🙂

```suggestion
  return Values((name) => 
    indexOfName.set(name, indexOfName.size + startIndex).get(name)
  )
```

Reply:
Thanks I'll do that now ✌🏼